android/sdl_android/src/main/java/com/smartdevicelink/util/SdlAppInfo.java

```diff
@@ -243,39 +248,20 @@ public String toString() {
      * @return true if vehicle type is supported.
      */
     public static boolean checkIfVehicleSupported(List<VehicleType> supportedVehicleList, VehicleType connectedVehicle) {
-        if (supportedVehicleList == null || supportedVehicleList.isEmpty() || connectedVehicle == null || connectedVehicle.getStore().isEmpty()) {
+        if (supportedVehicleList == null || supportedVehicleList.isEmpty() || connectedVehicle == null || connectedVehicle.getStore() == null|| connectedVehicle.getStore().isEmpty()) {
             return true;
         }
         if (supportedVehicleList.contains(connectedVehicle)) {
             return true;
         }
         for (VehicleType supportedVehicle: supportedVehicleList) {
-            String supportedVehicleMake = supportedVehicle.getMake();
-            String connectedVehicleMake = connectedVehicle.getMake();
-            if (supportedVehicleMake != null && connectedVehicleMake != null && connectedVehicleMake.equalsIgnoreCase(supportedVehicleMake)) {
-                String supportedVehicleModel = supportedVehicle.getModel();
-                String connectedVehicleModel = connectedVehicle.getModel();
-                if (supportedVehicleModel != null && connectedVehicleModel != null) {
-                    if (connectedVehicleModel.equalsIgnoreCase(supportedVehicleModel)) {
-                        boolean ret = true;
-                        String supportedVehicleModelYear = supportedVehicle.getModelYear();
-                        String connectedVehicleModelYear = connectedVehicle.getModelYear();
-                        if (supportedVehicleModelYear != null && connectedVehicleModelYear != null) {
-                            ret = connectedVehicleModelYear.equalsIgnoreCase(supportedVehicleModelYear);
-                        }
-                        String supportedVehicleTrim = supportedVehicle.getTrim();
-                        String connectedVehicleTrim = connectedVehicle.getTrim();
-                        if (supportedVehicleTrim != null && connectedVehicleTrim != null) {
-                            ret &= connectedVehicleTrim.equalsIgnoreCase(supportedVehicleTrim);
-                        }
-                        if (ret) {
-                            return true;
-                        }
-                    }
-                }
-                else {
-                    // Return true if only make is defined and it matches
-                    return true;
+            boolean areVehicleMakesEqual = CompareUtils.areStringsEqual(supportedVehicle.getMake(), connectedVehicle.getMake(), true, false);
```

Comment:
`public class CompareUtils {

    public static boolean areStringsEqual(String string1, String string2, boolean ignoreCase, boolean nullIsEqual) {

        if (string1 == null) {
            if (string2 == null) {
                return nullIsEqual;
            } else {
                return false;
            }
        } else {
            //At least String 1 is not null, use it for the remaining checks
            if (ignoreCase) {
                return string1.equalsIgnoreCase(string2);
            } else {
                return string1.equals(string2);
            }
        }
    }

}`